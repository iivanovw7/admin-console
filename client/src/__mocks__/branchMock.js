export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};