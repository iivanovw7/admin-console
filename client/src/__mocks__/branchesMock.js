export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
