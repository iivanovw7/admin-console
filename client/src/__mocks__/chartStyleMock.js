export const style = {
  type: "PieChart",
  id: 2
};
