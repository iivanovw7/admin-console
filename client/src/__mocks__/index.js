export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
