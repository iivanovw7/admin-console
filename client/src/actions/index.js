export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';