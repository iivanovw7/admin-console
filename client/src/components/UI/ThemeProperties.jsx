import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

