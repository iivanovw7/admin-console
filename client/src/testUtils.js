import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};