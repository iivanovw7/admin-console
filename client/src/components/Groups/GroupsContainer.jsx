import { Link, Paper, Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { changeGroupStatus, deleteGroup, getGroups, getSingleGroup } from '../../actions';
import Warning from '../UI/Dialogs/Warning';
import AlertSnackbar from '../UI/Notifications/Snackbar';
import { Container } from '../UI/ThemeProperties';
import { actionButton } from './GroupsActionButton';
import { deleteButton } from './GroupsDeleteButton';

const GroupsContainer = props => {
  const { classes, history, dispatch, groups, page, limit } = props;
  const [confDialog, dialogState] = useState(false);
  const [groupToDelete, setToDelete] = useState(null);
  const currPage = page || 1;
  const currLimit = limit || 8;

  useEffect(() => {
    dialogState(false);
    setToDelete(null);
  }, []);

  function displayStatus(status) {
    return status ? 'Active' : 'Disabled';
  }

  function handleGroupClick(id) {
    dispatch(getSingleGroup(id, history));
  }

  function handleDeleteAction(id) {
    setToDelete(id);
    dialogState(true);
  }

  function DialogueWindow() {
    return (
      <Warning
        opened={confDialog}
        close={() => {
          dialogState(false);
        }}
        message={'All users of removed group will be assigned to "Other" group.'}
        request={() => {
          dialogState(false);
          dispatch(deleteGroup(groupToDelete, history));
        }}
        mainText={'You are about to DELETE group! Are you sure ?'}
      />
    );
  }

  //Triggers notification if there are any messages in props
  function displayNotifications() {
    if (props.errorMessage && !props.messageConfirmed) {
      return displayNotification(props.errorMessage, false);
    }
    if (props.successMessage && !props.messageConfirmed) {
      return displayNotification(props.successMessage, true);
    }
  }

  //Returns notification container
  const displayNotification = (message, success) => (
    <AlertSnackbar
      dispatch={dispatch}
      message={message}
      afterConfirm={() => {
        dispatch(getGroups(currPage, currLimit, history));
      }}
      success={success}
    />
  );

  //Function returns control buttons according to current elements state
  function showActionButton(row) {
    if (row.status) {
      return actionButton('Disable', 'secondary', row, classes, dispatch);
    }
    return actionButton('Activate', 'primary', row, classes, dispatch);
  }

  //Function returns delete button with parameters
  function showDeleteButton(row) {
    return deleteButton(row, classes, handleDeleteAction);
  }

  return (
    <Paper className={classes.root}>
      <Table className={classes.tables}>
        <TableHead>
          <TableRow>
            <TableCell className={classes.groupNameCell}>
              <h3>Name</h3>
            </TableCell>
            <TableCell className={classes.groupStatusCell} align="center">
              <h3>Status</h3>
            </TableCell>
            <TableCell className={classes.groupDesktopCell} align="center"/>
            <TableCell className={classes.groupDesktopCell} align="center"/>
          </TableRow>
        </TableHead>
        <TableBody>
          {groups.map(row => (
            <TableRow key={row._id} className={classes.groupRowClass}>
              <TableCell component="th" scope="row" className={classes.groupNameCell}>
                <Link
                  component="button"
                  variant="body2"
                  onClick={() => {
                    handleGroupClick(row._id);
                  }}
                >
                  {row.name}
                </Link>
              </TableCell>
              <TableCell className={classes.groupStatusCell} align="center">
                <strong>
                  {displayStatus(row.status)}
                </strong>
              </TableCell>
              <TableCell className={classes.groupDesktopCell} align="center">
                {showActionButton(row)}
              </TableCell>
              <TableCell className={classes.groupDesktopCell} align="center">
                {showDeleteButton(row)}
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
      {DialogueWindow()}
      {displayNotifications()}
    </Paper>
  );
};

GroupsContainer.propTypes = {
  classes: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
    errorMessage: state.groups.error,
    successMessage: state.groups.success,
    messageConfirmed: state.groups.confirmed
  };
}

export default connect(mapStateToProps, {
  getSingleGroup,
  deleteGroup,
  changeGroupStatus
})(withStyles(Container)(withRouter(GroupsContainer)));
