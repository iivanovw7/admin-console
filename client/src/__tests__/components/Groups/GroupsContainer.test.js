import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { MemoryRouter, withRouter } from 'react-router-dom';
import * as mocks from '../../../__mocks__/';
import { getSingleGroup } from '../../../actions';
import GroupsContainer from '../../../components/Groups/GroupsContainer';
import { Container } from '../../../components/UI/ThemeProperties';
import { storeFactory } from '../../../testUtils';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);
  const history = createMemoryHistory('/dashboard');

  const Composition = connect(
    null, { getSingleGroup }
  )(withStyles(Container, { withTheme: true })(withRouter(GroupsContainer)));

  const props = {
    groups: mocks.GROUPS.output,
    history: history
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders groups container with styles
   */
  describe('Groups container', () => {
    it('Should render groups container with fields and styles', () => {
      wrapper = mount(
        <Provider store={store}>
          <MemoryRouter>
            <Composition {...props} />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.length).toEqual(1);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();
