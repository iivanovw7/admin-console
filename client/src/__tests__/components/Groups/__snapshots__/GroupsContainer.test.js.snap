// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups container Should render groups container with fields and styles 1`] = `"<div class=\\"MuiPaper-root-53 MuiPaper-elevation2-57 MuiPaper-rounded-54 withRouter-GroupsContainer--root-27 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----root-1\\"><table class=\\"MuiTable-root-80 withRouter-GroupsContainer--tables-33 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----tables-7\\"><thead class=\\"MuiTableHead-root-81\\"><tr class=\\"MuiTableRow-root-82 MuiTableRow-head-85\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 withRouter-GroupsContainer--groupNameCell-41 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupNameCell-15\\" scope=\\"col\\"><h3>Name</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupStatusCell-42 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupStatusCell-16\\" scope=\\"col\\"><h3>Status</h3></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\" scope=\\"col\\"></th><th class=\\"MuiTableCell-root-87 MuiTableCell-head-88 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\" scope=\\"col\\"></th></tr></thead><tbody class=\\"MuiTableBody-root-99\\"><tr class=\\"MuiTableRow-root-82 withRouter-GroupsContainer--groupRowClass-40 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupRowClass-14\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-GroupsContainer--groupNameCell-41 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupNameCell-15\\" scope=\\"row\\"><button class=\\"MuiTypography-root-105 MuiTypography-body2-113 MuiTypography-colorPrimary-135 MuiTypography-inline-140 MuiLink-root-100 MuiLink-button-104 MuiLink-underlineHover-102\\">Front-End Pool</button></th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupStatusCell-42 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupStatusCell-16\\"><strong>Active</strong></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\"><button class=\\"MuiButtonBase-root-167 MuiButton-root-141 MuiButton-text-143 MuiButton-textSecondary-145 MuiButton-flat-146 MuiButton-flatSecondary-148 MuiButton-sizeSmall-164\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-142\\">Disable</span><span class=\\"MuiTouchRipple-root-186\\"></span></button></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\"><button class=\\"MuiButtonBase-root-167 MuiButton-root-141 MuiButton-contained-152 MuiButton-containedSecondary-154 MuiButton-raised-155 MuiButton-raisedSecondary-157 MuiButton-sizeSmall-164\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-142\\">Delete</span><span class=\\"MuiTouchRipple-root-186\\"></span></button></td></tr><tr class=\\"MuiTableRow-root-82 withRouter-GroupsContainer--groupRowClass-40 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupRowClass-14\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-GroupsContainer--groupNameCell-41 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupNameCell-15\\" scope=\\"row\\"><button class=\\"MuiTypography-root-105 MuiTypography-body2-113 MuiTypography-colorPrimary-135 MuiTypography-inline-140 MuiLink-root-100 MuiLink-button-104 MuiLink-underlineHover-102\\">Back-End Pool</button></th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupStatusCell-42 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupStatusCell-16\\"><strong>Disabled</strong></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\"><button class=\\"MuiButtonBase-root-167 MuiButton-root-141 MuiButton-text-143 MuiButton-textPrimary-144 MuiButton-flat-146 MuiButton-flatPrimary-147 MuiButton-sizeSmall-164\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-142\\">Activate</span><span class=\\"MuiTouchRipple-root-186\\"></span></button></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\"><button class=\\"MuiButtonBase-root-167 MuiButton-root-141 MuiButton-contained-152 MuiButton-containedSecondary-154 MuiButton-raised-155 MuiButton-raisedSecondary-157 MuiButton-sizeSmall-164\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-142\\">Delete</span><span class=\\"MuiTouchRipple-root-186\\"></span></button></td></tr><tr class=\\"MuiTableRow-root-82 withRouter-GroupsContainer--groupRowClass-40 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupRowClass-14\\"><th class=\\"MuiTableCell-root-87 MuiTableCell-body-89 withRouter-GroupsContainer--groupNameCell-41 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupNameCell-15\\" scope=\\"row\\"><button class=\\"MuiTypography-root-105 MuiTypography-body2-113 MuiTypography-colorPrimary-135 MuiTypography-inline-140 MuiLink-root-100 MuiLink-button-104 MuiLink-underlineHover-102\\">TEST</button></th><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupStatusCell-42 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupStatusCell-16\\"><strong>Disabled</strong></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\"><button class=\\"MuiButtonBase-root-167 MuiButton-root-141 MuiButton-text-143 MuiButton-textPrimary-144 MuiButton-flat-146 MuiButton-flatPrimary-147 MuiButton-sizeSmall-164\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-142\\">Activate</span><span class=\\"MuiTouchRipple-root-186\\"></span></button></td><td class=\\"MuiTableCell-root-87 MuiTableCell-body-89 MuiTableCell-alignCenter-96 withRouter-GroupsContainer--groupDesktopCell-43 withRouter-Connect-WithStyles-withRouter-GroupsContainer-----groupDesktopCell-17\\"><button class=\\"MuiButtonBase-root-167 MuiButton-root-141 MuiButton-contained-152 MuiButton-containedSecondary-154 MuiButton-raised-155 MuiButton-raisedSecondary-157 MuiButton-sizeSmall-164\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label-142\\">Delete</span><span class=\\"MuiTouchRipple-root-186\\"></span></button></td></tr></tbody></table><div></div></div>"`;
